src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -688,9 +688,11 @@ ReactDOMComponent.Mixin = {
           }
           propValue = CSSPropertyOperations.createMarkupForStyles(propValue);
         }
-        var markup = null;
+        var markup = '';
         if (this._tag != null && isCustomComponent(this._tag, props)) {
-          markup = DOMPropertyOperations.createMarkupForCustomAttribute(propKey, propValue);
+          if (propKey != 'children' || typeof propValue === 'string' || typeof propValue === 'number') {
```

Comment:
To get past the closure compiler issue you can use `var CHILDREN = keyOf({children: null})` and then use that in place of the string literal.

That's the safest thing, though `children` might actually be in the DOM externs.